### PR TITLE
chore: Add snyk security scan

### PR DESCRIPTION
**Pull request by warber**
#### What this PR does / Why we need it:
This PR adds the snyk github action to scan for vulnerabilities when opening a PR.
This does not (yet) contain pushing the tag to snyk.io

### CODE REVIEW

**Review by arthurpitman — DISMISSED** [review]
LGTM

**Review by arthurpitman — APPROVED** [review]

**Review by Laubi — APPROVED** [review]

File: .github/workflows/snyk.yml
Comment (UnseenWizzard) [inline .github/workflows/snyk.yml:5]: how does snyk determine the version it is pushing the report for? Will these be reported for main or will we have a snyk entry for each branch (that we'll have to clean up later)

**Review by UnseenWizzard — APPROVED** [review]

File: .github/workflows/snyk.yml
Comment (UnseenWizzard) [reply .github/workflows/snyk.yml:5]: approved as I'm fine if we find this out after the action is merged, but we should take a look before we consider this fully done

**Pull request merged**
